task/sast-unicode-check/0.1/sast-unicode-check.yaml

```diff
@@ -0,0 +1,274 @@
+                return 1
+            else
+                echo "Message: Succeed to check usage of find-unicode-control" >&2
+            fi
+        }
+
+        # Find unicode control
+        unicontrol_scan()
+        {
+            # Split the arguments into an array
+            args=("${FIND_UNICODE_CONTROL_ARGS}")
+
+            LANG=en_US.utf8 ./find-unicode-control/find_unicode_control.py "${args[@]}" "${SOURCE_CODE_DIR}/source" \
+                >raw_sast_unicode_check_out.txt \
+                2>raw_sast_unicode_check_out.log \
+                || FUC_EXIT_CODE=$?
+
+            if [[ -z "${FUC_EXIT_CODE}" || "${FUC_EXIT_CODE}" -eq 1 ]]; then
+                # No non-printable unicode characters found, or successfully found
+                echo "Message: Succeed to run find-unicode-control command" >&2
+            else
+                # Failed to run the command
+                echo "Error: failed to run find unicode control command" >&2
+                return 1
+            fi
+
+            # Translate the output format
+            sed -i raw_sast_unicode_check_out.txt -E -e 's|(.*:[0-9]+)(.*)|\1: warning:\2|' -e 's|^|Error: UNICONTROL_WARNING:\n|'
+            status=$?
+            if [ "${status}" -ne 0 ] && [ "${status}" -ne 1 ]; then
+                echo "Error: failed to translate the unicontrol output format" >&2
+                return 1
+            else
+                echo "Message: Succeed to translate the unicontrol output format" >&2
+            fi
+        }
+
+        # Process all results as configured with CSGERP_OPTS
+        process_all_results()
+        {
+            CSGERP_OPTS=(
+                --mode=json
+                --remove-duplicates
+                --embed-context=3
+                --set-scan-prop="${SCAN_PROP}"
+                --strip-path-prefix="${SOURCE_CODE_DIR}"/source/
+            )
+            # In order to generate csdiff/v1, we need to add the whole path of the source code as sast-unicode-check only provides an URI to embed the context
+            csgrep "${CSGERP_OPTS[@]}" raw_sast_unicode_check_out.txt > processed_sast_unicode_check_out.json
+
+            csgrep --mode=evtstat processed_sast_unicode_check_out.json
+        }
+
+        # Filter known false positive
+        filter_kfp()
+        {
+            echo "Message: Filtering false positives in results files using csfilter-kfp..." >&2
+            CMD=(
+                csfilter-kfp
+                --verbose 
+                --kfp-git-url="${KFP_GIT_URL}"
+            )
+
+            if [[ -n "${PROJECT_NVR}" ]] && [[ "${RECORD_EXCLUDED}" == "true" ]]; then
+                CMD+=(
+                    --project-nvr="${PROJECT_NVR}"
+                    --record-excluded=raw-excluded-findings.json
+                )
+            fi
+
+            "${CMD[@]}" processed_sast_unicode_check_out.json  > sast_unicode_check_out.json
+            status=$?
```

Comment:
See my previous comment.  The `status=$?` will not work with `set -e` because non-zero exit code of the previous command will terminate the shell.

Reply:
Updated